src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/AppTestBase.cs

```diff
@@ -23,35 +23,52 @@ protected AppTestBase(ITestOutputHelper output, SharedBuildPerTestClassFixture b
     protected string Id { get; set; }
     protected string LogPath { get; set; }
 
-    protected void CopyTestAsset(string assetName, string generatedProjectNamePrefix = null)
+    protected void CopyTestAsset(string assetName, string generatedProjectNamePrefix = null, string? projectDirSuffix = null)
     {
         Id = $"{generatedProjectNamePrefix ?? assetName}_{GetRandomId()}";
         InitBlazorWasmProjectDir(Id);
 
         LogPath = Path.Combine(s_buildEnv.LogRootPath, Id);
         Utils.DirectoryCopy(Path.Combine(BuildEnvironment.TestAssetsPath, assetName), Path.Combine(_projectDir!));
 
-        switch(assetName)
+        if (!string.IsNullOrEmpty(projectDirSuffix))
         {
-            case "WasmBasicTestApp":
-                // WasmBasicTestApp consists of App + Library projects
-                _projectDir = Path.Combine(_projectDir!, "App");
-                break;
-            case "BlazorHostedApp":
-                // BlazorHostedApp consists of BlazorHosted.Client and BlazorHosted.Server projects
-                _projectDir = Path.Combine(_projectDir!, "BlazorHosted.Server");
-                break;
+             _projectDir = Path.Combine(_projectDir, projectDirSuffix);
         }
     }
 
+    protected void BuildAspNetCoreServingWASM(
+        string config,
+        string assetName,
+        string projectDirSuffix,
+        string aspNetCoreAppDir,
+        string? generatedProjectNamePrefix = null,
+        RuntimeVariant runtimeType = RuntimeVariant.SingleThreaded)
+    {
+        CopyTestAsset(assetName, generatedProjectNamePrefix, projectDirSuffix);
+        PublishProject(configuration: config,
```

Comment:
Just curious: Did you try to reference wasm project from aspnetcore the same way as blazor? Does it work? The underlaying SDK bits should be the same enough to work...